torch/_inductor/runtime/compile_tasks.py

```diff
@@ -0,0 +1,68 @@
+from __future__ import annotations
+
+import functools
+import os
+import sys
+import warnings
+from types import ModuleType
+from typing import Any, Callable
+
+
+def _reload_triton_kernel_in_subproc(reload_module, kernel_name):
+    return _module_to_triton_kernel(reload_module(), kernel_name)
+
+
+def _module_to_triton_kernel(mod, kernel_name):
+    kernel = getattr(mod, kernel_name)
+    kernel._reload_in_subproc = functools.partial(
+        _reload_triton_kernel_in_subproc,
+        mod._reload_in_subproc,
+        kernel_name,
+    )
+    return kernel
+
+
+def _reload_python_module_in_subproc(key, path):
+    codecache = sys.modules.get("torch._inductor.codecache")
+    if codecache:
+        return codecache.PyCodeCache.load_by_key_path(key, path)
+    else:
+        return _reload_python_module(key, path)
```

Comment:
Nit: AFAICT, this PR is mostly just moving methods around. This is the one place I don't quite grok. What's happening that requires the sys.modules.get? So a comment about what's going on here would be nice.

Reply:
It is so that in the next PR when we do:
```
sys.modules["torch"] = MockTorch()
```
we don't try to go via codecache.